### PR TITLE
feat(workflows): Automate stable release tags

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

Generating a new tag will also start the release pipeline.
Monday was chosen as it's 1 day after dependabot runs and is before the week starts.

### CODE REVIEW

File: .github/workflows/release-stable.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: release-stable
 
 on:
   push:
-    tags: ["v[1-9]+.[0-9]+.[0-9]+"]
+    tags: ["v[0-9]+.[0-9]+.[0-9]+"]
```
Comment (nderjung) [inline .github/workflows/release-stable.yaml]: Weird, how did this work before? :D

Comment (craciunoiuc) [reply .github/workflows/release-stable.yaml]: It did not 🪄 

I had to do it by hand 😢

Comment (craciunoiuc) [reply .github/workflows/release-stable.yaml]: either that or we pushed to stable and that picked it up

What I was doing was to generate a tag + also push to stable and that would pick it up

**Review by nderjung — COMMENTED** [review]
I wonder, this should actually also check if there is actually a diff between staging/HEAD and stable. If we have a holiday or there is a week where nothing changes, a new version will come out regardless. 

The pipeline should also merge staging into stable, this is part of the general release mechanism.

**Comment by craciunoiuc** [comment]
I can add some more checks, sure
Things left to do:
* Checkout full staging branch
* Check if commit difference between latest release and the new one is bigger than 0
* Push to stable branch
* Disable releases on stable push

Edit: Fixed all here

File: .github/workflows/tag-stable-weekly.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          NEW_TAG="v${array[0]}.${array[1]}.${array[2]}"
+
+          git checkout staging
+          git tag -a "$NEW_TAG" -m "KraftKit $NEW_TAG"
+          git push origin "$NEW_TAG"
+
+          # Rebase stable on staging
+          git checkout stable
+          git rebase staging
+          git push origin stable
```
Comment (nderjung) [inline .github/workflows/tag-stable-weekly.yaml]: can you reverse the order here: merge and push to stable, then tag

Comment (craciunoiuc) [reply .github/workflows/tag-stable-weekly.yaml]: oke

**Review by nderjung — APPROVED** [review]
Thanks!

Reviewed-by: Alexander Jung <alex@unikraft.io>
Approved-by: Alexander Jung <alex@unikraft.io>

**Pull request merged**